book/src/storage/transfer-to-object.md

```diff
@@ -1,4 +1,132 @@
+[Programmable Transaction Block (PTB)](../concepts/what-is-a-transaction.md). The `receive` (or
+`public_receive`) function must be called in order to exchange the `Receiving` instance for the
+actual object.
+
+While this may seem overwhelming, we will demonstrate the feature in a simple example.
+
+## Example
+
+To demonstrate the feature, let's consider a simple `PostOffice` object that can receive any objects
+transferred to it. Given that the transfer itself can be performed on the PTB level, or in a Move
+function, we will focus on the receiving part.
+
+```move
+module book::post_office;
+
+// While the `sui::transfer` module is implicitly imported, the `Receiving` type
+// is not. We need to import it explicitly.
+use sui::transfer::Receiving;
+
+/// The `PostOffice` object that can receive any objects transferred to it via
+/// a custom `receive` function.
+public struct PostOffice has key {
+    id: UID,
+}
+
+/// Voila! The `receive` function can now be called on the `PostOffice` object
+/// to receive any objects transferred to it.
+public fun public_receive<T: key + store>(
+    po: &mut PostOffice,
+    to_receive: Receiving<T>
+): T {
+    transfer::public_receive(&mut po.id, to_receive)
+}
+
+/// For objects without `store` (though, they would require special handling), we
+/// can implement the non-public version of the `receive` function.
+public fun receive<T: key>(po: &mut PostOffice, to_receive: Receiving<T>): T {
+    transfer::receive(&mut po.id, to_receive)
+}
+```
+
+In this example, we have defined a `PostOffice` object that can receive any objects transferred to
+it. We omitted the creation of the `PostOffice` for brevity; additionally, an implementation like
+this would require some authorization to prevent unauthorized claims. We talk about authorization
+patterns in the [Advanced Programmability](./../programmability/) chapter.
+
+## Limitations
+
+Transfer to Object must be used with caution, as the recipient object must be implemented in a way
+that allows receiving. Worth noting, that objects owned by other objects must be taken by value, and
+cannot be borrowed as references, unlike objects owned by accounts.
+
+Transfer to Object is not the only way to create a relationship between objects. There are more
+flexible and feature-rich ways, such as the
+[Dynamic Fields](./../programmability/dynamic-fields.md), which we explore in the
+[Advanced Programmability](./../programmability/) chapter.
+
+## Summary
+
+- Objects can be transferred to other objects by calling the `transfer` function.
+- The recipient object must be able to `receive` transferred objects, otherwise the objects may be
+  lost.
+- The `Receiving` type is a special type that references an object-owned object. It is passed as a
+  special input to a transaction and cannot be constructed dynamically.
+- Once received, the object can be used normally, including being transferred again, even to the
```

Comment:
I don't think the "including being transferred again" part was discussed above, so it probably shouldn't be mentioned in the summary.